utils/doclint/generateDotnetApi.js

```diff
@@ -0,0 +1,695 @@
+  });
+
+  if (memberKind === 'argument') {
+    if (['params', 'event'].includes(name)) { // just in case we want to add others
+      return `@${name}`;
+    } else {
+      return name;
+    }
+  }
+  // check if there's an alias in the docs, in which case
+  // we return that, otherwise, we apply our dotnet magic to it
+  if (member) {
+    if (member.alias !== name) {
+      return member.alias;
+    }
+  }
+
+  let assumedName = name.charAt(0).toUpperCase() + name.substring(1);
+
+  switch (memberKind) {
+    case "interface":
+      // apply name mapping if the map exists
+      let mappedName = classNameMap ? classNameMap.get(assumedName) : null;
+      if (mappedName)
+        return mappedName;
+      return `I${assumedName}`;
+    case "method":
+      if (member && member.async)
+        return `${assumedName}Async`;
+      return assumedName;
+    case "event":
+      return `${assumedName}`;
+    case "enum":
+      return `${assumedName}`;
+    default:
+      return `${assumedName}`;
+  }
+}
+
+/**
+ * 
+ * @param {Documentation.Member} member 
+ * @param {Documentation.Class|Documentation.Type} parent 
+ * @param {string[]} out
+ */
+function renderMember(member, parent, out) {
+  let output = line => {
+    if (typeof (line) === 'string')
+      out.push(`\t${line}`);
+    else
+      out.push(...line);
+  }
+
+  let name = translateMemberName(member.kind, member.name, member);
+  if (member.kind === 'method') {
+    renderMethod(member, parent, output, name);
+  } else {
+    let type = translateType(member.type, parent, (t) => generateNameDefault(member, name, t, parent));
+    if (member.kind === 'event') {
+      if (!member.type)
+        throw `No Event Type for ${name} in ${parent.name}`;
+      if (member.spec)
+        output(XmlDoc.renderXmlDoc(member.spec, maxDocumentationColumnWidth).map(x => `\t${x}`));
+      if (parent && (classNameMap.get(parent.name) === type))
+        output(`event EventHandler ${name};`); // event sender will be the type, so we're fine to ignore
+      else
+        output(`event EventHandler<${type}> ${name};`);
+    } else if (member.kind === 'property') {
+      if (member.spec)
+        output(XmlDoc.renderXmlDoc(member.spec, maxDocumentationColumnWidth).map(x => `\t${x}`));
+      output(`${type} ${name} { get; set; }`);
+    } else {
+      throw `Problem rendering a member: ${type} - ${name} (${member.kind})`;
+    }
+  }
+
+  // we're separating each entry and removing the final blank line when rendering
+  out.push('');
+}
+
+/**
+ * 
+ * @param {Documentation.Member} member 
+ * @param {string} name 
+ * @param {Documentation.Type} t 
+ * @param {*} parent 
+ */
+function generateNameDefault(member, name, t, parent) {
+  if (!t.properties
+    && !t.templates
+    && !t.union
+    && t.expression === '[Object]')
+    return 'object';
+
+  // we'd get this call for enums, primarily
+  let enumName = generateEnumNameIfApplicable(member, name, t, parent);
```

Comment:
How about this:

```js
if (enumName)
  return enumName;

if (!member)
  return t.name;
```

to simplify remaining part of the method?

Reply:
Yeah, I've added the handling when `name` exists. I _sort of_ want to keep the rest, just in case, but also, probably best to just `throw` if an enum without a name happens...